src/main/scala/is/hail/io/LoadMatrix.scala

```diff
@@ -23,160 +23,47 @@ object LoadMatrix {
   }
 
   // this assumes that col IDs are in last line of header.
-  def parseHeader(line: String, sep: String = "\t", hasRowIDName: Boolean = false): Array[String] =
-    if (hasRowIDName)
-      line.split(sep).tail
-    else
-      line.split(sep)
+  def parseHeader(line: String, sep: String = "\t", nAnnotations: Int, annotationHeaders: Option[Seq[String]]): (Array[String], Array[String]) =
+    annotationHeaders match {
+      case None =>
+        val r = line.split(sep)
+        (r.slice(0, nAnnotations), r.slice(nAnnotations, r.length))
+      case Some(h) =>
+        assert(h.length == nAnnotations, s"Expected $nAnnotations column annotations, but only ${ h.length } names provided as column headers")
```

Comment:
This should be a fatal since it can be user-forced.

Reply:
if I get rid of nAnnotations and do the check in python, is it okay to leave the assert here?